apps/cowswap-frontend/src/legacy/state/orders/middleware/appziMiddleware.ts

```diff
@@ -105,13 +105,22 @@ function _triggerNps(
     return
   }
 
-  openNpsAppziSometimes({
-    ...npsParams,
-    secondsSinceOpen: timeSinceInSeconds(openSince),
-    explorerUrl,
-    chainId,
-    orderType: uiOrderType,
-  })
+  triggerAppziSurvey(
+    {
+      ...npsParams,
+      secondsSinceOpen: timeSinceInSeconds(openSince),
+      explorerUrl,
+      chainId,
+      orderType: uiOrderType,
+      account: order?.owner,
+      pendingOrderIds: getPendingOrderIds(store, chainId).join(','),
+    },
+    uiOrderType === UiOrderType.LIMIT ? 'limit' : 'nps' // Different survey exclusive for limit orders
```

Comment:
nit: maybe make a function so that comment is not longer required

i.e.
`getSurveyName(uiOrderType): SurveyType`

Reply:
Maybe this could be a util inside `libs/common-utils/src/appzi.ts` ?